### PR TITLE
Openpulse drawer

### LINKED ISSUE #1920: Pulse visualizations
### What is the expected enhancement?
For ease of experiment development, a full suite of visualization tools should accompany the pulse API. This should include visualizations for pulses, channels and schedules. See #1713. For more information.

This will be implemented in #1902.

### PR DESCRIPTION
**Pull request by taalexander**
### Summary
This module adds a `pulse_drawer` interface to terra that operates on `Schedule`s,  `Instruction`s and `SamplePulse`s. Submitting this on the behalf of @nkanazawa1989 who did the hard work. This is submitted in place of the previous stale PR #1902. This PR is based off of #2186 and should only be merged after it.

Closes #1920.

### Details and comments

See `tests.visualization.pulse_visualization` for example code.

Example of a schedule. 

schedule_matplotlib_ref

Example of an plotted pulse.

pulse_matplotlib_ref

Example of a plotted instruction.

instruction_matplotlib_ref

### CODE REVIEW

File: qiskit/visualization/pulse_visualization.py [hunk 1]
```diff
@@ -15,96 +15,67 @@
...
-        filename (str): Name required to save pulse image.
+        data (ScheduleComponent or SamplePulse): Data to plot
+        dt (float): Time interval of samples
+        style (OPStylePulse or OPStyleSched): A style sheet to configure plot appearance
+        filename (str): Name required to save pulse image
+        output (str): Select the output method to use for drawing the pulse object.
+            Valid choices are `mpl`.
+        interp_method (Callable): interpolation function
+            See `qiskit.visualization.interpolation` for more information
+        scaling (float): scaling of waveform amplitude
```
Comment (nonhermitian) [inline qiskit/visualization/pulse_visualization.py]: Aren't all pulse amplitudes in [0,1]?

**Comment by nonhermitian** [comment]
Also, the copyright headers need to be updated

File: qiskit/visualization/interpolation.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+"""
+interpolation module for pulse visualization.
+"""
+from functools import partial
+
+from scipy import interpolate
+import numpy as np
+
+
+def interp1d(time, samples, nop, kind='linear'):
```
[the remark this answers is not in the document]
Comment (taalexander) [reply qiskit/visualization/interpolation.py]: No it looks really poor with square pulses. It was there by default and was very confusing.

**Comment by taalexander** [comment]
Thanks for the review @nonhermitian. In general, I am going to hold off on any major modifications as this is @nkanazawa1989  code and he is away. 

`The circuit_latex_ref.png and circuit_matplotlib_ref.pngfiles do not see to match for things like barriers and id gates. Is this the expected behavior?` I don't know what you mean by this? There are no barriers or id gates for pulse.

File: qiskit/visualization/pulse_visualization.py (hunk 1 above)
Comment (taalexander) [reply qiskit/visualization/pulse_visualization.py]: Should be. I'll need to look into what this does.

**Comment by nonhermitian** [comment]
Ok yes, it seems the png in question are just being renamed.

**Comment by taalexander** [comment]
Thank you @nonhermitian and @mtreinish. I agree the layout could be improved. I should point out that this PR also sat waiting to be reviewed since the beginning of March (#1902), so there was plenty of time to comment (on this particular issue, not the rest of pulse that is *very* last minute). I just wanted to point this out on the behalf of @nkanazawa1989 since he is not around.

**Pull request merged**